libs/platforms/rosa/terraform/terraform.py

```diff
@@ -0,0 +1,262 @@
+        if cleanup_code == 0:
+            self.logging.debug(
+                f"Confirm cluster {cluster_name} deleted by attempting to describe the cluster. This should fail if the cluster is removed."
+            )
+            check_code, check_out, check_err = self.utils.subprocess_exec(
+                "rosa describe cluster -c " + cluster_name, log_output=False
+            )
+            if check_code != 0:
+                cluster_info["status"] = "deleted"
+            else:
+                cluster_info["status"] = "not deleted"
+        else:
+            cluster_info["status"] = "not deleted"
+        cluster_end_time = int(datetime.datetime.utcnow().timestamp())
+        cluster_info["destroy_duration"] = cluster_delete_end_time - cluster_start_time
+        cluster_info["destroy_all_duration"] = cluster_end_time - cluster_start_time
+        try:
+            with open(cluster_info['path'] + "/metadata_destroy.json", "w") as metadata_file:
+                json.dump(cluster_info, metadata_file)
+        except Exception as err:
+            self.logging.error(err)
+            self.logging.error(f"Failed to write metadata_install.json file located at {cluster_info['path']}")
+        if self.es is not None:
+            cluster_info["timestamp"] = datetime.datetime.utcnow().isoformat()
+            self.es.index_metadata(cluster_info)
+
+    def get_workers_ready(self, kubeconfig, cluster_name):
+        super().get_workers_ready(kubeconfig, cluster_name)
+        myenv = os.environ.copy()
+        myenv["KUBECONFIG"] = kubeconfig
+        self.logging.info(f"Getting node information for Terraform installed cluster {cluster_name}")
+        nodes_code, nodes_out, nodes_err = self.utils.subprocess_exec("oc get nodes -o json", extra_params={"env": myenv, "universal_newlines": True}, log_output=False)
+        try:
+            nodes_json = json.loads(nodes_out)
+        except Exception as err:
+            self.logging.debug(f"Cannot load command result for cluster {cluster_name}")
+            self.logging.debug(err)
+            return 0
+        nodes = nodes_json["items"] if "items" in nodes_json else []
+        status = []
+        for node in nodes:
+            labels = node.get("metadata", {}).get("labels", {})
+            if "node-role.kubernetes.io/worker" in labels and "node-role.kubernetes.io/control-plane" not in labels and "node-role.kubernetes.io/infra" not in labels:
+                conditions = node.get("status", {}).get("conditions", [])
+                for condition in conditions:
+                    if "type" in condition and condition["type"] == "Ready":
+                        status.append(condition["status"])
+        status_list = {i: status.count(i) for i in status}
+        ready_nodes = status_list["True"] if "True" in status_list else 0
+        return ready_nodes
+
+    def create_cluster(self, platform, cluster_name):
+        super().create_cluster(platform, cluster_name)
```

Comment:
how would it work - u execute TF apply multiple times? If so, this is not the way to add load here. You should use count= variable and define it in the TF yaml. And TF will do all the scale creation for you

Reply:
In rosa-burner, we have the bits to do concurrency with batches, each request spins up a thread and apply TF so we can control how we want to scale, 
for ex, to create 10 clusters, 
10 req at a time - 1 sec interval = would spin all 10 at the same time
1 req at a time - 1 sec interval = would spin sequentially with a sec pause
We follow this for hypertshift work, David and myself intentionally exclude the `count` variable in to the script as it might add extra level of concurrency that we can't control or watch. But if you think you want that specifically we can consider doing that option as well.

Reply:
This wont cover the real use case, since this is not how TF users will use it. You can do both count and batches within TF configuration.

Reply:
We already saw in the past when MOBB team started deploying labs using TF, OCM service had issues, while, I assume, it behaved fine under rosa burner load.

Reply:
Okay, if that is the preferred mechanism , lets use TF `count`. @morenod lets discuss to make use of TF [parallelism](https://developer.hashicorp.com/terraform/cli/commands/apply#parallelism-n) and count.

Reply:
Interesting, are users using tf parallelism to create a number of clusters at the same time? I would imagine doing multiply tf applies is representative of different users all trying to create a cluster using tf.

Reply:
I would say we need both. On different accounts/vpcs...

Reply:
@bardielle can you upload some example using the count to the examples/ folder of the module project? Im trying to use it but I have a lot of interlocks on the operator roles module and the output.tf, and I dont see anywhere an example about which is the correct way to use the count on the rhcs_cluster_rosa_classic module

Reply:
ok, I think I found the correct way to add the count. @bardielle can you check if the new tf files are according to requirements about creating any number of clusters?

Reply:
@rfreiman@redhat.com I am afraid that adding `count` logic into rosa-burner is not as easy as I thought, using TF count would make it harder to track and record per cluster duration & status. Our deployment model in threads will not support nested TF concurrency, as it is not easy to detect failures while it bursts that many request and hard to find the cause. We have been running 1-2 request per minute and that itself a very aggressive rate than a real-world situation. 
I don't know how TF provider handles count and parallelism, @dbarda@redhat.com do you have a working template with count var that was tested ? would be easy to play with it
If it needs to be done that way, then we'd prefer a simple bash script to do that.

Reply:
But this is the real usecase that ocm will have to deal with, right? Customers won't have rosa burner - they'll use tf to deploy clusters at scale. We already had it during redhat event and ocm was chocking

Reply:
I apologize for the delayed response.
[Here](https://github.com/amalykhi/terraform-provider-rhcs/blob/multicount/examples/create_rosa_sts_cluster/oidc_configuration/cluster_with_managed_oidc_config/main.tf) you can find an example of using the `count` attribute that was published by @amalykhi

You should specify the count attribute within the declarations for all resources and modules

Reply:
Thanks @bardielle, @romfreiman we scale test real clusters at a slower rate as we had experienced AWS throttle limits before, we arrived this number 1-2 per minute based on the telemetry inflight requests rate(they are even lower), but happy to find find the new limits during this test. 
We also run OCM api load test(10 per sec) with fake cluster creates and that had higher failure rate(non 200s), so we arrived to 2 per second for fake clusters. [Archived report](https://docs.google.com/document/d/1tZWH2Rf9p1tVd99vyxvj1XIJzVMDG5MP3Ghy-kvNLnU/edit#heading=h.60v4i2a4e7to). I presume this would be lesser with real clusters due to AWS provisioning bits. 
But as Sai mentioned creating many clusters using a single tf template would not represent a real world multi user load, curious to know the difference that TF provider would bring to OCM api with single vs many tf template. (Apologies for the lack TF knowledge), we can replicate that using rosa-burner as well with many templates firing concurrently.

Reply:
Based on [this](https://redhat-internal.slack.com/archives/C03RNFXM72M/p1697998158905239), using `count` seems to be a practical requirement, apparently we have to keep the OCM api limits while running this test. 
Also we always run the test using a single AWS account, we follow this model for the benefit of quota/billing allocations based on past experience. 
We will work on this proposed idea in a different script or a fork to maintain existing hosted-cp logic intact, we can freeze this PR with single cluster per TF template(without `count`)